File: 프로그래머스/1/12901. 2016년/2016년.cpp
#include <string>
#include <vector>

using namespace std;

string solution(int a, int b) {
    string answer = "";
    
    int date[] = { 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    vector<string> day = { "FRI", "SAT", "SUN", "MON", "TUE", "WED", "THU" };
    
    int cal_date = 0;
    for(int i=1; i<a; i++){
        cal_date += date[i-1];
    }
    
    cal_date += b-1;
    cal_date %= day.size();
    
    answer = day[cal_date];
    
    return answer;
}

File: 프로그래머스/1/12901. 2016년/2016년.js
function solution(a, b) {
    var answer = '';
    
    let date = [ 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 ];
    let day = [ "FRI", "SAT", "SUN", "MON", "TUE", "WED", "THU" ];
    
    let sum = 0;
    for(let i=1; i<a; i++){
        sum += date[i-1];
    }
    
    sum += b-1;
    sum %= day.length;
    
    answer = day[sum];
    return answer;
}